packages/dtslint/src/checks.ts

```diff
@@ -56,16 +56,24 @@ export async function checkPackageJson(dirPath: string, typesVersions: readonly
   }
 }
 
+/**
+ * numbers in `CompilerOptions` might be enum values mapped from strings
+ */
+export type CompilerOptionsRaw = {
+  [K in keyof CompilerOptions]?: CompilerOptions[K] extends number | undefined
+    ? string | number | undefined
```

Comment:
do properties normally get literal union types instead of `string | number | undefined` ?

Reply:
Most of the ones that are getting transformed here are enum types, which are basically literal union types.